src/services/expense.service.js

```diff
@@ -0,0 +1,80 @@
+const { Op } = require('sequelize');
+const { Expense } = require('../models/Expense.model.js');
+
+const getExpenses = async ({ userId, categoryList, from, to }) => {
+  const whereClause = {};
+
+  if (userId) {
+    whereClause.userId = userId;
+  }
+
+  if (categoryList && categoryList.length > 0) {
+    whereClause.category = { [Op.in]: categoryList };
+  }
+
+  if (from) {
+    whereClause.spentAt = { [Op.gte]: new Date(from) };
+  }
+
+  if (to) {
+    whereClause.spentAt = { ...whereClause.spentAt, [Op.lte]: new Date(to) };
+  }
+
+  const expenses = await Expense.findAll({
+    where: whereClause,
+  });
+
+  return expenses;
+};
+
+const createExpense = ({
+  userId,
+  spentAt,
+  title,
+  amount,
+  category = '',
+  note = '',
+}) => {
+  return Expense.create({
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  });
+};
+
+const getExpenseById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const updateExpense = async ({ id, data }) => {
+  const [numberOfAffectedRows] = await Expense.update(data, { where: { id } });
```

Comment:
The `update` method returns an array where the first element is the number of affected rows. Ensure that the `data` object contains valid fields for updating an expense to avoid potential issues with no rows being affected.